### PR TITLE
CHE-4768. Fix receiving file tracking operation calls from client at refactoring

### PR DESCRIPTION
**Pull request by RomanNikitenko**
Signed-off-by: Roman Nikitenko <rnikitenko@codenvy.com>

### What does this PR do?
1. Fix receiving file tracking operation calls on server side at refactoring.
2. Send file tracking operation calls on server side asynchronously to avoid incorrect "External operation" events 

### What issues does this PR fix or reference?
#4768

#### Changelog
Fix receiving file tracking operation calls from client at refactoring

### CODE REVIEW

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/resources/impl/ResourceManager.java [hunk 1]
```diff
@@ -492,51 +496,61 @@ private NewProjectConfigDto asDto(MutableProjectConfig config) {
                 deletedFilesController.add(source.getLocation().toString());
             }
 
-            eventBus.fireEvent(newFileTrackingSuspendEvent());
-
-            store.dispose(source.getLocation(), !source.isFile()); //TODO: need to be tested
+            return clientServerEventService.sendFileTrackingSuspendEvent().thenPromise(success -> {
+                eventBus.fireEvent(newFileTrackingSuspendedEvent());
```
Comment (vzhukovs) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/resources/impl/ResourceManager.java]: Do we really need this event bus event then?

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/resources/impl/ResourceManager.java [hunk 2]
```diff
@@ -492,51 +496,61 @@ private NewProjectConfigDto asDto(MutableProjectConfig config) {
...
+                store.dispose(source.getLocation(), !source.isFile()); //TODO: need to be tested
+
+                return ps.move(source.getLocation(), destination.parent(), destination.lastSegment(), force)
+                         .thenPromise(ignored -> {
+                             if (source.isProject() && source.getLocation().segmentCount() == 1) {
+                                 return ps.getProjects().then((Function<List<ProjectConfigDto>, Resource>)updatedConfigs -> {
+                                     clientServerEventService.sendFileTrackingResumeEvent().then(arg -> {
+                                         eventBus.fireEvent(newFileTrackingResumedEvent());
```
Comment (vzhukovs) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/resources/impl/ResourceManager.java]: The same as above

**Comment by RomanNikitenko** [comment]
@vzhukovskii  You are right - we can avoid to use FileTrackingEvent on client side. I've deprecated this class and updated the pull request. Thank you!

**Review by vzhukovs — APPROVED** [review]

**Comment by RomanNikitenko** [comment]
@vparfonov I've fixed conflicts and updated the pull request

**Review by vparfonov — APPROVED** [review]

**Pull request merged**